### PR TITLE
Add AddChainWithDeadline

### PR DESCRIPTION
**Pull request by rolandshoemaker**
Adds new method that unblocks work on letsencrypt/boulder#1365.

(Probably should have tests... hmm)

### CODE REVIEW

File: go/client/logclient.go [hunk 1]
```diff
@@ -242,12 +257,18 @@ func (c *LogClient) addChainWithRetry(path string, chain []ct.ASN1Cert) (*ct.Sig
...
 
 // AddPreChain adds the (DER represented) Precertificate |chain| to the log.
 func (c *LogClient) AddPreChain(chain []ct.ASN1Cert) (*ct.SignedCertificateTimestamp, error) {
-	return c.addChainWithRetry(AddPreChainPath, chain)
+	return c.addChainWithRetry(AddPreChainPath, chain, nil)
+}
+
+// AddChainWithDeadline adds the (DER represented) X509 |chain| to the log and
+// fails if the provided deadline is passed before the chain is submitted.
+func (c *LogClient) AddChainWithDeadline(chain []ct.ASN1Cert, deadline *time.Timer) (*ct.SignedCertificateTimestamp, error) {
```
Comment (jmhodges) [inline go/client/logclient.go]: Why not just make this take a context.Context?

**Comment by rolandshoemaker** [comment]
Switched to `context.Context`

File: go/client/logclient.go [hunk 2]
```diff
@@ -179,20 +180,36 @@ func (c *LogClient) postAndParse(uri string, req interface{}, res interface{}) (
...
 	for _, link := range chain {
 		req.Chain = append(req.Chain, base64.StdEncoding.EncodeToString(link))
 	}
-	done := false
 	httpStatus := "Unknown"
+	backoffSeconds := 0
+	done := false
 	for !done {
-		backoffSeconds := 0
+		backoffTimer := time.NewTimer(0)
```
Comment (jmhodges) [inline go/client/logclient.go:195]: So, why the switch to NewTimer from a time.Sleep?

Comment (rolandshoemaker) [inline go/client/logclient.go:195]: Timer provides a channel which can be checked concurrently with the
context, unlike a Sleep which just blocks everything

**Comment by jmhodges** [comment]
LGTM with the context made as the first parameter (in line with most go code that uses them elsewhere, I believe) and renaming `deadline` to `ctx`

**Comment by rolandshoemaker** [comment]
Doooone.

**Comment by jmhodges** [comment]
LGTM

**Pull request merged**